watermill/watermill.go

```diff
@@ -58,18 +59,18 @@ func NewSubscriber(connection string, logger watermill.LoggerAdapter) (message.S
 	var c message.Subscriber
 	switch u.Scheme {
 	case "mysql", "oracle", "postgres":
-		db, err := sql.Open(connection)
+		path, db, err := sqlUri.Open(connection)
 		if err != nil {
 			return nil, err
 		}
-		sub := sqlplugin.Subscriber{DB: db}
+		sub := sqlplugin.Subscriber{DB: db, SelectPath: path}
 		c, err = sub.NewSubscriber(nil, logger)
 		if err != nil {
 			return nil, err
 		}
```

Comment:
Let's move this to additional func in sql package, so that the logic here is only in selecting what to build and not how